dataland-e2etests/src/test/kotlin/org/dataland/e2etests/utils/ApiAccessor.kt

```diff
@@ -38,80 +53,149 @@ class ApiAccessor {
         SfdrDataControllerApi(BASE_PATH_TO_DATALAND_BACKEND)
     val testDataProviderForSfdrData =
-        TestDataProvider(SfdrData::class.java)
+        FrameworkTestDataProvider(SfdrData::class.java)
     val sfdrUploaderFunction = { companyId: String, sfdrData: SfdrData ->
         val companyAssociatedSfdrData = CompanyAssociatedDataSfdrData(companyId, sfdrData)
         dataControllerApiForSfdrData.postCompanyAssociatedSfdrData(
             companyAssociatedSfdrData
         )
     }
 
-    fun <T> uploadCompanyAndFrameworkData(
+    val dataControllerApiForSmeData =
+        SmeDataControllerApi(BASE_PATH_TO_DATALAND_BACKEND)
+    val testDataProviderForSmeData =
+        FrameworkTestDataProvider(SmeData::class.java)
+    val smeUploaderFunction = { companyId: String, smeData: SmeData ->
+        val companyAssociatedSmeData = CompanyAssociatedDataSmeData(companyId, smeData)
+        dataControllerApiForSmeData.postCompanyAssociatedSmeData(
+            companyAssociatedSmeData
+        )
+    }
+
+    fun <T> uploadCompanyAndFrameworkDataForOneFramework(
         listOfCompanyInformation: List<CompanyInformation>,
         listOfFrameworkData: List<T>,
         frameworkDataUploadFunction: (companyId: String, frameworkData: T) -> DataMetaInformation
-    ) {
-        if (listOfCompanyInformation.size == listOfFrameworkData.size) {
-            tokenHandler.obtainTokenForUserType(TokenHandler.UserType.Uploader)
-            listOfCompanyInformation.forEachIndexed { index, companyInformation ->
-                val receivedCompanyId = companyDataControllerApi.postCompany(companyInformation).companyId
-                frameworkDataUploadFunction(receivedCompanyId, listOfFrameworkData[index])
+    ): List<UploadInfo> {
+        val listOfUploadInfo: MutableList<UploadInfo> = mutableListOf()
+        tokenHandler.obtainTokenForUserType(TokenHandler.UserType.Uploader)
+        listOfCompanyInformation.forEach { companyInformation ->
+            val receivedStoredCompany = companyDataControllerApi.postCompany(companyInformation)
+            listOfFrameworkData.forEach { frameworkDataSet ->
+                val receivedDataMetaInformation =
+                    frameworkDataUploadFunction(receivedStoredCompany.companyId, frameworkDataSet)
+                listOfUploadInfo.add(
+                    UploadInfo(
+                        companyInformation,
+                        receivedStoredCompany,
+                        receivedDataMetaInformation
+                    )
+                )
             }
-        } else {
-            throw throw IllegalArgumentException(
-                "The list of companyInformation has ${listOfCompanyInformation.size} elements, " +
-                    "and the list of frameworkData has ${listOfFrameworkData.size} elements. " +
-                    "The numbers of elements of both lists need to match."
+        }
+        return listOfUploadInfo
+    }
+
+    @Suppress("kotlin:S138")
```

Comment:
Wolltest du die Funktion nicht kürzer machen?

Reply:
Ich habe gestern nach unserem Team-Call nach dem Daily noch etwas experimentiert, und auch nochmal mit Philip gesprochen.
Hierbei hat sich herausgestellt, dass es aktuell nicht möglich ist, auf das Mapping hier in der Form zu verzichten, oder es als "Iteration" mit einem Einzeiler abzuarbeiten.
An irgendeiner Stelle MUSS dieses Mapping zwingend passieren, und ich habe es auch schon so kurz wie möglich gehalten.
Daher haben wir uns geeinigt, dass wir diese eine Ausnahme hier machen müssen. Diese Funktion wird wohl oder übel mit jedem weiteren Framework wachsen.
Irgendwann in Zukunft müssen/können wir das evtl. anders umsetzen.